src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java

```diff
@@ -613,7 +613,7 @@ public void setPreviousImprisonmentStatusToInactive(final LocalDateTime expiryTi
 
     public LegalStatus getLegalStatus() {
         return getActiveImprisonmentStatus().map(
-            is -> is.getImprisonmentStatus().getLegalStatus()
+            is -> is.getImprisonmentStatus() != null ? is.getImprisonmentStatus().getLegalStatus() : null
```

Comment:
There are some prisoners with imprisonment statuses not on the table `IMPRISONMENT_STATUSES`. The old endpoint just returns null for these, so shall we.